src/pages/trails/treasuries/components/TreasuryTokenEditDialog.vue

```diff
@@ -0,0 +1,78 @@
+<script>
+import { mapActions } from 'vuex';
+import TreasuryTokenSettingsEdit from './TreasuryTokenSettingsEdit';
+
+export default {
+  name: 'TreasuryTokenEditDialog',
+  components: {
+    TreasuryTokenSettingsEdit
+  },
+  props: {
+    show: { type: Boolean, required: true },
+    treasury: { type: Object, required: true },
+  },
+  data() {
+    return {
+      settings: [],
+      submitting: false,
+    };
+  },
+  watch: {
+    show() {
+      if (this.show) {
+        this.settings = this.treasury.settings.map(x => ({key:x.key, value:!!x.value}));
+      }
+    }
+  },
+  methods: {
+    ...mapActions('trails', ['editTreasurySettings']),
+    async onSaveSettings() {
+      this.submitting = true;
+      const success = await this.editTreasurySettings({
+        settings: this.settings.map(x => ({key:x.key, value:x.value?1:0})),
+        treasury: this.treasury
+      });
+      this.submitting = false;
+      if (success) {
+        this.showSuccessMsg('Token features updated successfully');
+        this.$emit('close');
+      } else {
+        this.showSuccessMsg('An error has ocurred');
+      }
+    },
+  },
+};
+</script>
+
+<template lang="pug">
```

Comment:
How would you feel about changing this to html since its a new file?

Reply:
Yes, of course. Fixed here [fcec55c](https://github.com/telosnetwork/app-telos-native/pull/224/commits/fcec55c351ecae17f26547bf79fc6c7f0e6d855c)